rlclientlib/live_model_impl.cc

```diff
@@ -417,4 +423,35 @@ namespace reinforcement_learning {
     return error_code::success;
   }
 
+  int post_process_rank(ranking_response& response, decision_mode decision_mode) {
+    if (decision_mode == IMITATION_MODE) {
+      std::vector<action_prob> copied_action_prob;
+
+      for (auto it = response.begin(); it != response.end(); ++it) {
+        copied_action_prob.push_back(*it);
+      }
+
+      std::sort(copied_action_prob.begin(), copied_action_prob.end(), [](const action_prob& a, const action_prob& b) {
+        return a.action_id < b.action_id;
+        }
+      );
+
+      response.clear_ranking();
+      response.set_chosen_action_id(copied_action_prob.begin()->action_id);
+      for (auto it = copied_action_prob.begin(); it != copied_action_prob.end(); ++it) {
+        action_prob cur = *it;
+        response.push_back(cur.action_id, cur.probability);
+      }
+    }
+    return error_code::success;
+  }
+
+  decision_mode to_decision_modes(const char* decision_mode) {
+    if (std::strcmp(decision_mode, "IMITATION_MODE") == 0) {
```

Comment:
>IMITATION_MODE [](start = 36, length = 14)

move this to constants.h?

Reply:
Fixed.

---
In reply to: [386552274](https://github.com/VowpalWabbit/reinforcement_learning/pull/129#discussion_r386552274) [](ancestors = 386552274)